### ISSUE TITLE
does not lint anymore

### ISSUE THREAD

**Issue by despairblue** [ricardofbarros/linter-js-standard#30]
Happens when updating to version 2.0.0 or later

1.2.0 still lints.

**Comment by ricardofbarros** [comment]
Did you update AtomLinter as well?

**Comment by despairblue** [comment]
atom linter 1.2.4

**Comment by despairblue** [comment]
It's probably because of the two new options, which I have both disabled.

**Comment by ricardofbarros** [comment]
If so you still have the style option fallback, 

what can happen is if you have `checkStyleDevDependencies` option ticked it won't lint the file if the there isn't specified the style of the project on package.json `devDependencies`

**Comment by despairblue** [comment]
I haven't ticket `checkStyleDevDependencies` nor `honorStyleSettings` and my chosen style is standard.

My config looks like
```JSON
  "linter-js-standard": {}
```

Settings:
![screenshot from 2015-07-23 01-25-09](https://cloud.githubusercontent.com/assets/927609/8839802/ba02a430-30d9-11e5-96cc-8341a54711e6.png)

package.json:
```JSON
  "devDependencies": {
    "faucet": "0.0.1",
    "groc": "0.6.3",
    "istanbul": "0.3.0",
    "jasmine": "2.2.1",
    "node-inspector": "0.10.1",
    "nodemon": "1.3.2",
    "standard": "*",
    "supertest": "0.13.0",
    "tape": "^4.0.0"
  }
```

**Comment by despairblue** [comment]
This is probably as puzzling to you as it is to me :confused:

**Comment by ricardofbarros** [comment]
Hmm awkward, let me try to replicate.

**Comment by despairblue** [comment]
A I get a rejected promise:
```
Error: Notification must be created with string message: undefined
  at Notification.module.exports.Notification.validate (/usr/share/atom/resources/app.asar/src/notification.js:25:15)
  at new Notification (/usr/share/atom/resources/app.asar/src/notification.js:20:12)
  at NotificationManager.module.exports.NotificationManager.addError (/usr/share/atom/resources/app.asar/src/notification-manager.js:41:35)
  at /home/despairblue/.atom/packages/linter/lib/editor-linter.coffee:78:30
```

**Comment by despairblue** [comment]
The rejection does not happen with other linters.

**Comment by despairblue** [comment]
[this line](https://github.com/atom-community/linter/blob/v1.2.4/lib/editor-linter.coffee#L78) ends up being called with this string as `error`
```javascript
"standard: Use JavaScript Standard Style (https://github.com/feross/standard)
"
```

Thus `error.message` is undefined.

**Comment by ricardofbarros** [comment]
What the hell?? :anguished:

**Comment by ricardofbarros** [comment]
Could you insert in this [line](https://github.com/ricardofbarros/linter-js-standard/blob/master/lib/utils/linter.js#L47)
```js
console.log(occurences)
```

**Comment by despairblue** [comment]
Get's never called. :(

**Comment by despairblue** [comment]
Which makes sense, since [this](https://github.com/ricardofbarros/linter-js-standard/blob/master/lib/linter-js-standard.js#L35) promise is rejected, so `then` is executed.

**Comment by despairblue** [comment]
Ok standard outputs `standard: Use JavaScript Standard Style (https://github.com/feross/standard)` to stderr and all subsequent errors to stdout. The problem is [this](https://github.com/AtomLinter/atom-linter/blob/master/lib/helpers.coffee#L19) part of atom-linter.

If you pass no options it assumes 'stdout' and will [reject](https://github.com/AtomLinter/atom-linter/blob/master/lib/helpers.coffee#L28) if there is anything in stderr (which there is: `standard: Use JavaScript Standard Style (https://github.com/feross/standard)`). If you pass `stderr` as option (or anything but stdout or null/undefined for that matter) it will [resolve](https://github.com/AtomLinter/atom-linter/blob/master/lib/helpers.coffee#L32) with stderr (`standard: Use JavaScript Standard Style (https://github.com/feross/standard)`).

This is a pretty fucked up situation, there is simply no way to reach line 30 :confused: 

https://github.com/AtomLinter/atom-linter/commit/40d9f0efcae1ab61302cc3b90a75385c31966578 is the offending commit.

**Comment by despairblue** [comment]
As precaution I'd always use fixed version in the package.json.

**Comment by ricardofbarros** [comment]
Couldn't we redirect stderr to stdout?

**Comment by ricardofbarros** [comment]
Try the following fix in this [line](https://github.com/ricardofbarros/linter-js-standard/blob/master/lib/linter-js-standard.js#L34) to see if it works for you:

```js
args.push('2>&1')
```

**Comment by ricardofbarros** [comment]
In the upcoming release that promise will have a `.catch`

**Comment by despairblue** [comment]
This does nothing, I doubt it's passed to a shell as such, the process is probably started with a syscall, which would mean the argument is just passed to through to `standard`.

**Comment by despairblue** [comment]
I made a PR to fix this for now.

**Comment by despairblue** [comment]
btw, all this makes me think of

![workflow](https://imgs.xkcd.com/comics/workflow.png)

**Comment by ricardofbarros** [comment]
Ehehe :smile: 

Thank you @despairblue

**Comment by ricardofbarros** [comment]
Published a patch, could you see if that works for you?

**Comment by despairblue** [comment]
:+1: works. @steelbrain is also implementing an option to disable the rejection when the linter writes to stderr.

I'll issue another PR when it's published.

**Comment by ricardofbarros** [comment]
Cool! Many thanks :+1:

**Comment by ricardofbarros** [comment]
atom-linter released 3.0.0 with it came a new option you were talking about `throwOnStdErr`, I released the 2.2.0 to follow with these changes.
